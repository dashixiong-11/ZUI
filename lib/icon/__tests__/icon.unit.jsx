import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import {mount} from 'enzyme'

describe('icon', () => {
   it('render', () => {
      const json = renderer.create(<Icon/>).toJSON()
      expect(json).toMatchSnapshot()
   })

   it('onClick', () => {
      const fn = jest.fn()
      const c = mount(<Icon name='alipay' onClick={ fn } />)
      c.find('svg').simulate('click')
      expect(fn).toBeCalled()
   })
})

