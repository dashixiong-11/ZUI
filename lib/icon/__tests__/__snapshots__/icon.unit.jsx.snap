// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render 1`] = `
<svg
  className="zui-icon"
  style={
    Object {
      "fill": "undefined",
      "fontSize": "undefinedpx",
    }
  }
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
